azurerm/resource_arm_virtual_hub_test.go

```diff
@@ -0,0 +1,294 @@
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMVirtualHub_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-network-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_wan" "test" {
+  name                = "acctest-VirtualWan-%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+}
+
+resource "azurerm_virtual_hub" "test" {
+  name                = "acctest-VirtualHub-%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+  address_prefix      = "10.0.1.0/24"
+  virtual_wan_id      = "${azurerm_virtual_wan.test.id}"
+}
+`, rInt, location, rInt, rInt)
+}
+
+func testAccAzureRMVirtualHub_requiresImport(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+resource "azurerm_virtual_hub" "import" {
+  name                = "${azurerm_virtual_hub.test.name}"
+  location            = "${azurerm_virtual_hub.test.location}"
+  resource_group_name = "${azurerm_virtual_hub.test.name}"
+}
+}
+`, testAccAzureRMVirtualHub_basic(rInt, location))
+}
+
+func testAccAzureRMVirtualHub_complete(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-network-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctest-VirtualNetwork-%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+  address_space       = ["10.5.0.0/16"]
+}
+
+resource "azurerm_network_security_group" "test" {
+  name                = "acctest-NetworkSecurityGroup-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_subnet" "test" {
+  name                      = "acctest-Subnet-%d"
+  resource_group_name       = "${azurerm_resource_group.test.name}"
+  virtual_network_name      = "${azurerm_virtual_network.test.name}"
+  address_prefix            = "10.5.1.0/24"
+  network_security_group_id = "${azurerm_network_security_group.test.id}"
+}
+
+resource "azurerm_virtual_wan" "test" {
+  name                = "acctest-VirtualWan-%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+}
+
+resource "azurerm_virtual_hub" "test" {
+  name                = "acctest-VirtualHub-%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+  address_prefix      = "10.0.2.0/24"
+
+  virtual_wan_id  = "${azurerm_virtual_wan.test.id}"
+
+  virtual_network_connection {
+	name                                       = "testConnection"	
```

Comment:
We should fix the spacing here so it lines up with the attributes below

Reply:
fixed